### ISSUE TITLE
Inputs created from XmlDocument with "PreserveWhitespace = true" generate unexpected differences.

### ISSUE THREAD

**Issue by jlocans** [xmlunit/xmlunit.net#38]
Hello!

First of all, thanks for the great library!

Maybe this is intentional, but I just spent way too much time figuring out why any 3 of the whitespace ignore rules were not working and were in fact returning twice the differences than w/o any additional rules.

Turns out, if you create `XmlDocument` with `PreserveWhitespace` set to `true` and use that as input source, then the whitespace rules don't work.

I've modified one of the test cases:
```csharp
[Test]
public void TestDiff_withIgnoreWhitespaces_shouldSucceed() {
    // prepare testData
    string controlXml = "<a><b>Test Value</b></a>";
    string testXml = "<a>\n <b>\n  Test Value\n </b>\n</a>";
    var controlDoc = new XmlDocument
    {
        PreserveWhitespace = true
    };
    controlDoc.LoadXml(controlXml);
    var testDoc = new XmlDocument
    {
        PreserveWhitespace = true
    };
    testDoc.LoadXml(testXml);

    // run test
    var myDiff = DiffBuilder.Compare(Input.FromDocument(controlDoc).Build())
        .WithTest(Input.FromDocument(testDoc).Build())
        .IgnoreWhitespace()
        .Build();

    // validate result
    Assert.IsFalse(myDiff.HasDifferences(), "XML similar " + myDiff.ToString());
}
```

I guess I would have expected that any of the whitespace ignore rules would take precedence over `PreserveWhitespace` setting.

**Comment by bodewig** [comment]
To be honest I would have expected the same as you. :-)

It doesn't work without `PreserveWhitespace` either. The problem is the parsed document contains `XmlWhitespace` nodes that the code in `Nodes` doesn't handle at all. Patch upcoming.

**Comment by jlocans** [comment]
Glad I could help.
> It doesn't work without PreserveWhitespace either

`IgnoreElementContentWhitespace` does seem to work though. In my test case, if document was created with preserving whitespace, then a simple compare w/o any filtering would return 8 `SIMILAR` differences. Adding `IgnoreElementContentWhitespace` would strangely generate 22 differences, including `DIFFERENT` differences.

However creating document w/o `PreserveWhitespace` in combination with `IgnoreElementContentWhitespace` returns 0 differences.

**Comment by bodewig** [comment]
you are correct, when I wrote what you quote my failing unit test actually contained `PreserveWhitespace = true`, I had just forgotten about it.

Anyway, this is fixed now and https://github.com/xmlunit/xmlunit.net/commit/5d5edaaeec1a6e3f2c7a27a119af1221cf756aee#diff-ee535be5744414e990a2353b3b7b3cc4d2ade8bb30e046c28405572e69f3a048R31 passes (it failed before my change).

**Comment by bodewig** [comment]
I've just published nuget packages for 2.9.2

**Comment by bodewig** [comment]
you're welcome.
